[CI] EsqlSpecIT test {csv-spec:k8s-timeseries-promql.day_of_week_sunday} failing

**Build Scans:**
- [elasticsearch-intake #37993 / part3](https://gradle-enterprise.elastic.co/s/ubkfmzhgcer4k)
- [elasticsearch-pull-request #138899 / part-3](https://gradle-enterprise.elastic.co/s/vctq7ekejg6me)
- [elasticsearch-pull-request #138912 / part-3](https://gradle-enterprise.elastic.co/s/uokf6zlknecz4)
- [elasticsearch-pull-request #138904 / part-3](https://gradle-enterprise.elastic.co/s/26fgg743fdbtw)
- [elasticsearch-intake #37994 / part3](https://gradle-enterprise.elastic.co/s/3z75keihde5oe)
- [elasticsearch-pull-request #138912 / part-3](https://gradle-enterprise.elastic.co/s/fy3enutue7fqe)
- [elasticsearch-pull-request #138904 / part-3](https://gradle-enterprise.elastic.co/s/ieturdji5gqj6)
- [elasticsearch-pull-request #138899 / part-3](https://gradle-enterprise.elastic.co/s/g52b5jmpgr6ig)
- [elasticsearch-intake #37989 / part3](https://gradle-enterprise.elastic.co/s/c6oxe5qktn3ee)
- [elasticsearch-pull-request #138897 / part-3](https://gradle-enterprise.elastic.co/s/l6mi7rndq3ria)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT" -Dtests.method="test {csv-spec:k8s-timeseries-promql.minute_scalar}" -Dtests.seed=A5D5CABF73B5496D -Dtests.locale=en-TT -Dtests.timezone=Europe/Mariehamn -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bcsv-spec%3Ak8s-timeseries-promql.day_of_week_sunday%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected more data but no more entries found after [0]
Data mismatch:

Actual:
result:double | step:datetime

Expected:
result:double | step:datetime
30.0          | 2024-05-10T14:30:00.000Z

```

**Issue Reasons:**
- [main] 35 consecutive failures in test test {csv-spec:k8s-timeseries-promql.day_of_week_sunday}
- [main] 7 consecutive failures in step part3
- [main] 28 consecutive failures in step part-3
- [main] 7 consecutive failures in pipeline elasticsearch-intake
- [main] 19 consecutive failures in pipeline elasticsearch-pull-request
- [main] 35 failures in test test {csv-spec:k8s-timeseries-promql.day_of_week_sunday} (100.0% fail rate in 35 executions)
- [main] 7 failures in step part3 (100.0% fail rate in 7 executions)
- [main] 28 failures in step part-3 (100.0% fail rate in 28 executions)
- [main] 7 failures in pipeline elasticsearch-intake (100.0% fail rate in 7 executions)
- [main] 19 failures in pipeline elasticsearch-pull-request (100.0% fail rate in 19 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -399,45 +399,9 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeApproximationIT
   method: test {csv-spec:mv_expand.explosionStats}
   issue: https://github.com/elastic/elasticsearch/issues/146939
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {csv-spec:k8s-timeseries-promql.day_of_week_sunday}
-  issue: https://github.com/elastic/elasticsearch/issues/146947
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {csv-spec:k8s-timeseries-promql.hour_scalar}
-  issue: https://github.com/elastic/elasticsearch/issues/146948
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-promql.hour_scalar}
-  issue: https://github.com/elastic/elasticsearch/issues/146949
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {csv-spec:k8s-timeseries-promql.minute_scalar}
-  issue: https://github.com/elastic/elasticsearch/issues/146951
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-promql.day_of_week_sunday}
-  issue: https://github.com/elastic/elasticsearch/issues/146952
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-promql.minute_scalar}
-  issue: https://github.com/elastic/elasticsearch/issues/146953
 - class: org.elasticsearch.xpack.ml.integration.DatafeedJobsIT
   method: testStopLookback_forceTrue_closeJobFalse
   issue: https://github.com/elastic/elasticsearch/issues/146955
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT
-  method: test {csv-spec:k8s-timeseries-promql.day_of_week_sunday}
-  issue: https://github.com/elastic/elasticsearch/issues/146947
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT
-  method: test {csv-spec:k8s-timeseries-promql.hour_scalar}
-  issue: https://github.com/elastic/elasticsearch/issues/146948
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT
-  method: test {csv-spec:k8s-timeseries-promql.minute_scalar}
-  issue: https://github.com/elastic/elasticsearch/issues/146951
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
-  method: test {csv-spec:k8s-timeseries-promql.minute_scalar}
-  issue: https://github.com/elastic/elasticsearch/issues/146956
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
-  method: test {csv-spec:k8s-timeseries-promql.hour_scalar}
-  issue: https://github.com/elastic/elasticsearch/issues/146957
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
-  method: test {csv-spec:k8s-timeseries-promql.day_of_week_sunday}
-  issue: https://github.com/elastic/elasticsearch/issues/146958
 - class: org.elasticsearch.xpack.inference.mapper.SemanticTextFieldMapperTests
   method: testMinimalToMaximal {p0=true p1=ENTERPRISE}
   issue: https://github.com/elastic/elasticsearch/issues/146966

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/k8s-timeseries-promql.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/k8s-timeseries-promql.csv-spec
@@ -547,10 +547,10 @@ result:double | step:datetime
 day_of_week_sunday
 required_capability: promql_command_v0
 required_capability: promql_time_functions
-PROMQL index=k8s step=1h start="2024-05-12T00:00:00.000Z" end="2024-05-12T01:00:00.000Z" result=(day_of_week());
+PROMQL index=k8s step=1h start="2024-05-10T00:00:00.000Z" end="2024-05-10T01:00:00.000Z" result=(day_of_week(vector(1715472000)));
 
 result:double | step:datetime
-0.0           | 2024-05-12T00:00:00.000Z
+0.0           | 2024-05-10T00:00:00.000Z
 ;
 
 day_of_year_scalar
@@ -565,19 +565,19 @@ result:double | step:datetime
 hour_scalar
 required_capability: promql_command_v0
 required_capability: promql_time_functions
-PROMQL index=k8s step=1h start="2024-05-10T14:00:00.000Z" end="2024-05-10T15:00:00.000Z" result=(hour());
+PROMQL index=k8s step=1h start="2024-05-10T00:00:00.000Z" end="2024-05-10T01:00:00.000Z" result=(hour());
 
 result:double | step:datetime
-14.0          | 2024-05-10T14:00:00.000Z
+0.0           | 2024-05-10T00:00:00.000Z
 ;
 
 minute_scalar
 required_capability: promql_command_v0
 required_capability: promql_time_functions
-PROMQL index=k8s step=1h start="2024-05-10T14:30:00.000Z" end="2024-05-10T15:30:00.000Z" result=(minute());
+PROMQL index=k8s step=1h start="2024-05-10T00:00:00.000Z" end="2024-05-10T01:00:00.000Z" result=(minute());
 
 result:double | step:datetime
-30.0          | 2024-05-10T14:30:00.000Z
+0.0           | 2024-05-10T00:00:00.000Z
 ;
 
 time_in_arithmetic

